### PR TITLE
Refactor/global target flags

### LINKED ISSUE #85: Introduce subcommands for gardenctl target
Currently, `gardenctl target` has no real subcommands for the different targeting types.
This comes with several drawbacks:
- Subcommands not listed in help
- No dedicated help for different subcommands
- complex code needs to differentiate between target types

After introducing the subcommands, we can also make the global target flags lokal, adding them only to the commands where they make sense. In case of target command, also differentiate between subcommands, as some of the target flags may not make sense for a specific target type. E.g. something like `gardenctl target shoot foo --shoot bar` should not be supported.

### PR DESCRIPTION
**Pull request by sven-petersen**
**What this PR does / why we need it**:

Instead of registering the target flags (`--garden`, `--shoot`, ...) on the root cmd they are now registered per command. This way the flags are only available where they have an effect.

**Which issue(s) this PR fixes**:
Fixes #85 (2nd step)

**Special notes for your reviewer**:

follow up: https://github.com/gardener/gardenctl-v2/issues/195 

**Release note**:

```other developer
Register target-flags only on cmds that require them instead of globally
```

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @sven-petersen for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: pkg/target/manager.go [hunk 1]
```diff
@@ -31,10 +32,28 @@ var (
...
 )
 
+// AutoCompletionsFuncs provides a set of functions that lists suitable values for
+// targeting while taking the currently targeted garden account.
+// In case the manager gets refactored/split up these functions should stay together.
+type AutoCompletionsFuncs interface {
+	// ShootNames returns all shoots for the current target.
+	ShootNames(ctx context.Context) ([]string, error)
+	// SeedNames returns all seeds for the current target.
+	// The current target must at least point to a garden.
```
Comment (petersutter) [inline pkg/target/manager.go:42]: > `// The current target must at least point to a garden.`

This applies all functions except `GardenNames`. Either we remove it here (as we hinted it already in the interface description or add it also for the other functions

File: pkg/target/manager_test.go [hunk 2]
```diff
@@ -612,4 +612,107 @@ var _ = Describe("Target Manager", func() {
 			})
 		})
 	})
+
+	Context("AutoCompletionsFuncs", func() {
+		// var clientConfig clientcmd.ClientConfig
```
Comment (petersutter) [inline pkg/target/manager_test.go:617]:
```suggestion
```

**Review by petersutter — APPROVED** [review]
/lgtm (but requires rebase)

File: pkg/flags/target.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/target"
+)
+
+func RegisterTargetFlagCompletionFuncs(cmd *cobra.Command, factory util.Factory, ioStreams util.IOStreams, flags *pflag.FlagSet) {
```
Comment (holgerkoser) [inline pkg/flags/target.go:21]: Rename to `RegisterTargetFlagsCompletionFuncs` or RegisterCompletionFuncsForTargetFlags`

**Review by holgerkoser — APPROVED** [review]
/lgtm
Do forget to rebase

**Pull request merged**